homeassistant/components/zcs_azzurro_inverter/zcs_azzurro_api.py

```diff
@@ -0,0 +1,137 @@
+        response = requests.post(
+            ZcsAzzurroApi.ENDPOINT,
+            headers=headers,
+            json=data,
+            timeout=ZcsAzzurroApi.REQUEST_TIMEOUT,
+        )
+        if response.status_code == 401:
+            raise HTTPError(f"{response.status_code}: Authentication Error")
+        return response
+
+    def realtime_data_request(
+        self,
+        required_values: list[str] | None = None,
+    ) -> dict:
+        """Request realtime data."""
+        if not required_values:
+            required_values = [ZcsAzzurroApi.REQUIRED_VALUES_ALL]
+        data = {
+            ZcsAzzurroApi.REALTIME_DATA_KEY: {
+                ZcsAzzurroApi.COMMAND_KEY: ZcsAzzurroApi.REALTIME_DATA_COMMAND,
+                ZcsAzzurroApi.PARAMS_KEY: {
+                    ZcsAzzurroApi.PARAMS_THING_KEY: self._thing_serial,
+                    ZcsAzzurroApi.PARAMS_REQUIRED_VALUES_KEY: ZcsAzzurroApi.REQUIRED_VALUES_SEP.join(
+                        required_values
+                    ),
+                },
+            }
+        }
+        response = self._post_request(data)
+        if not response.ok:
+            raise ConnectionError("Response did not return correctly")
+        response_data: dict[str, Any] = response.json()[ZcsAzzurroApi.REALTIME_DATA_KEY]
+        _LOGGER.debug("fetched realtime data %s", response_data)
+        if not response_data[ZcsAzzurroApi.RESPONSE_SUCCESS_KEY]:
+            raise ConnectionError("Response did not return correctly")
+        return response_data[ZcsAzzurroApi.PARAMS_KEY][
+            ZcsAzzurroApi.RESPONSE_VALUES_KEY
+        ][0][self._thing_serial]
+
+    def alarms_request(self) -> dict:
+        """Request alarms."""
+        required_values = [ZcsAzzurroApi.REQUIRED_VALUES_ALL]
+        data = {
+            ZcsAzzurroApi.DEVICES_ALARMS_KEY: {
+                ZcsAzzurroApi.COMMAND_KEY: ZcsAzzurroApi.DEVICES_ALARMS_COMMAND,
+                ZcsAzzurroApi.PARAMS_KEY: {
+                    ZcsAzzurroApi.PARAMS_THING_KEY: self._thing_serial,
+                    ZcsAzzurroApi.PARAMS_REQUIRED_VALUES_KEY: ZcsAzzurroApi.REQUIRED_VALUES_SEP.join(
+                        required_values
+                    ),
+                },
+            }
+        }
+        response = self._post_request(data)
+        if not response.ok:
+            raise ConnectionError("Response did not return correctly")
+        response_data: dict[str, Any] = response.json()[
+            ZcsAzzurroApi.DEVICES_ALARMS_KEY
+        ]
+        _LOGGER.debug("fetched realtime data %s", response_data)
+        if not response_data[ZcsAzzurroApi.RESPONSE_SUCCESS_KEY]:
+            raise ConnectionError("Response did not return correctly")
+        return response_data[ZcsAzzurroApi.PARAMS_KEY][
+            ZcsAzzurroApi.RESPONSE_VALUES_KEY
+        ][0][self._thing_serial]
+
+    @property
+    def identifier(self):
+        """object identifier."""
+        return f"{self.client}_{self._thing_serial}"
```

Comment:
You need to make this an external package.
https://developers.home-assistant.io/docs/creating_component_code_review?_highlight=pypi#4-communication-with-devicesservices

Reply:
I thought it was a [strong] suggestion, not a must (but I noticed I was wrong, reading it again from your link). Either it is just few calls. It will require me a little longer to fix this